clone.go

```diff
@@ -0,0 +1,262 @@
+			currentOverhang = overhangs[overhangIndex]
+			nextOverhang = overhangs[overhangIndex+1]
+			if enzyme.Directional == true {
+				if (currentOverhang.Forward == true) && (nextOverhang.Forward == false) {
+					fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+				}
+				if nextOverhang.Position > len(seq.Sequence) {
+					break
+				}
+			} else {
+				fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+				if nextOverhang.Position > len(seq.Sequence) {
+					break
+				}
+			}
+		}
+	}
+
+	// Convert fragment sequences into fragments
+	var fragments []Fragment
+	for _, fragment := range fragmentSeqs {
+		fragments = append(fragments, Fragment{Sequence: fragment[enzyme.EnzymeOverhangLen : len(fragment)-enzyme.EnzymeOverhangLen], ForwardOverhang: fragment[:enzyme.EnzymeOverhangLen], ReverseOverhang: fragment[len(fragment)-enzyme.EnzymeOverhangLen:]})
+	}
+	return fragments
+}
+
+func recurseLigate(wg *sync.WaitGroup, c chan string, seedFragment Fragment, fragmentList []Fragment) {
+	// Recurse ligate simulates all possible ligations of a series of fragments. Each possible combination begins with a "seed" that fragments from the pool can be added to.
+	defer wg.Done()
+	// If the seed ligates to itself, we can call it done with a successful circularization!
+	if seedFragment.ForwardOverhang == seedFragment.ReverseOverhang {
+		c <- seedFragment.ForwardOverhang + seedFragment.Sequence
+	} else {
+		for _, newFragment := range fragmentList {
+			// If the seedFragment's reverse overhang is ligates to a fragment's forward overhang, we can ligate those together and seed another ligation reaction
+			if seedFragment.ReverseOverhang == newFragment.ForwardOverhang {
+				newSeed := Fragment{seedFragment.Sequence + seedFragment.ReverseOverhang + newFragment.Sequence, seedFragment.ForwardOverhang, newFragment.ReverseOverhang}
+				wg.Add(1)
+				go recurseLigate(wg, c, newSeed, fragmentList)
+			}
+			// This checks if we can ligate the next fragment in its reverse direction. We have to be careful though - if our seed has a palindrome, it will ligate to itself
+			// like [-> <- -> <- -> ...] infinitely. We check for that case here as well.
+			if (seedFragment.ReverseOverhang == ReverseComplement(newFragment.ReverseOverhang)) && (seedFragment.ReverseOverhang != ReverseComplement(seedFragment.ReverseOverhang)) { // If the second statement isn't there, program will crash on palindromes
+				newSeed := Fragment{seedFragment.Sequence + seedFragment.ReverseOverhang + ReverseComplement(newFragment.Sequence), seedFragment.ForwardOverhang, ReverseComplement(newFragment.ForwardOverhang)}
+				wg.Add(1)
+				go recurseLigate(wg, c, newSeed, fragmentList)
+			}
+		}
+	}
+}
+
+// Ligate simulates ligation of all possible fragment combinations into circular plasmids.
+func Ligate(fragments []Fragment, maxClones int) []CloneSequence {
```

Comment:
```suggestion
// Ligate simulates ligation of all possible fragment combinations into circular plasmids.
func SimulateLigationOfAllPossibleFragments(fragments []Fragment, maxClones int) []CloneSequence {
```

Reply:
Gonna call it `CircularLigate`, since I think that'll make the most sense - since it's definitely not a linear ligate function.